cmd/babble/commands/keygen.go

```diff
@@ -0,0 +1,31 @@
+package commands
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/mosaicnetworks/babble/src/crypto"
+	"github.com/spf13/cobra"
+)
+
+// KeygenCmd produces a new key pair and prints it to stdout
+var KeygenCmd = &cobra.Command{
+	Use:   "keygen",
+	Short: "Dump new key pair",
+	RunE:  keygen,
+}
+
+func keygen(cmd *cobra.Command, args []string) error {
+	pemDump, err := crypto.GeneratePemKey()
+	if err != nil {
+		fmt.Println("Error generating PemDump")
+		os.Exit(2)
+	}
+
+	fmt.Println("PublicKey:")
+	fmt.Println(pemDump.PublicKey)
+	fmt.Println("PrivateKey:")
+	fmt.Println(pemDump.PrivateKey)
```

Comment:
This is not a good idea to print the PrivateKey into the tty (not a good idea to print it at all)
We should make that generation seamlessly at the first run if no key exists (cf the `Babble.Keygen()` method)
I wonder if we really need a `babble keygen` cli command considering this

Reply:
Agree that keys should be written to files, not stdout. However, I think the babble keygen command is useful for people who want to create key pairs without running a node. For example when someone is gathering public keys to produce a peers.json, they will ask all participants to generate their key pair and send over the public key. In fact, I am not sure how useful it is to generate a key at first run if no key exists; at least until we have dynamic participants. In the current situation, with static participants, the newly created key is not know by anyone but the node that just started; it is useless. But I can see that in dynamic participants it will be usefull to automatically create a key pair (if none exists) before sending a JoinRequest for example.